### DISCUSSION TITLE
Flickering on resize redraw

### DISCUSSION THREAD

**Discussion by sQVe** [kovidgoyal/kitty#4395] [Q&A] [answered]
👋🏼 

I'm trying out `kitty` and loving it so far but I'm getting pretty terrible redraws when issuing a resize. The main issue is text flashing and moving when in `neovim`. I'm running `i3` and very commonly spawn new terminals, just to do something quickly and throw it away, so it's essential to me that resizing looks OK.

I've recorded the issue and I'm curious if this is a common thing - or if there's something off with my setup? I'm running *Arch* with latest kernel and `mesa` drivers.

https://user-images.githubusercontent.com/2284724/147411598-9a4085e7-9fe2-43fd-ab85-806eb83d7741.mp4

As a comparison - this is what it looks like when using `alacritty`:

https://user-images.githubusercontent.com/2284724/147411721-64ad4ca8-4398-4112-ab5a-33d6d0cc4593.mp4

**Comment by kovidgoyal** [comment, marked as answer]
That will be because your editor is redrawing the screen and kitty is
fast enough to show you the redraws. Try it with some plain text at a
shell, where kitty itself does the redraw and you will see much less
flicker if any at all. To solve this your editor needs to implement
support for atomic screen redraws: https://gitlab.freedesktop.org/terminal-wg/specifications/-/merge_requests/2

**Reply by sQVe** [reply]
Thank you for quick response and all the work you've put into open source. 🙏🏼 ❤️ 

Do you by any chance know if this has been raised in `vim` or `neovim`?

**Comment by kovidgoyal** [comment]
On Sun, Dec 26, 2021 at 07:38:10AM -0800, Oskar Grunning wrote:
> Thank you for quick response and all the work you've put into open source. 🙏🏼 ❤️ 

You are welcome!

> 
> Do you by any chance know if this has been raised in `vim` or `neovim`? 

No idea, sorry.

**Comment by michaelfortunato** [comment]
Hi @sQVE, I'm having this issue too in my neovim window when adding or removing kitty windows (all on the same kitty tab). Definitely feels like an issue with neovim. Not sure if there is much we can do. I prefer using kitty "splits", where one window runs my editor neovim, and then other runs the shell, as opposed to using neovim's :terminal emulator and navigating windows within neovim because kitty is the best. I think the flicker in neovim is worth the trade off. I do not really know how this all works, but Kovid Goyal's explanation makes sense -- I guess that the kitty window containining neovim sends a SIGWINCH to neovim on resize but neovim (assuming) does not support atomic redraws and kitty is fast enough to show the redraws.
